src/mono/wasm/runtime-test.js

```diff
@@ -251,74 +304,42 @@ var Module = {
 			var path = asset.substr(Module.config.deploy_prefix.length);
 			writeContentToFile(content, path);
 			*/
-
-			if (typeof window != 'undefined') {
-				return fetch (asset, { credentials: 'same-origin' });
-			} else {
-				// The default mono_load_runtime_and_bcl defaults to using
-				// fetch to load the assets.  It also provides a way to set a
-				// fetch promise callback.
-				// Here we wrap the file read in a promise and fake a fetch response
-				// structure.
-				return new Promise ((resolve, reject) => {
-					var bytes = null, error = null;
-					try {
-						bytes = read (asset, 'binary');
-					} catch (exc) {
-						error = exc;
-					}
-					var response = { ok: (bytes && !error), url: asset,
-						arrayBuffer: function () {
-							return new Promise ((resolve2, reject2) => {
-								if (error)
-									reject2 (error);
-								else
-									resolve2 (new Uint8Array (bytes));
-						}
-					)}
-					}
-					resolve (response);
-				})
-			}
+			return IOHandler.fetch (asset, { credentials: 'same-origin' });
```

Comment:
Missing `.catch` ?

Reply:
I'm pretty sure that when this is used (library-mono.js line 1950) the surrounding try catch block handles this.
```javascript
try {
  if (asset.name === attemptUrl) {
    if (ctx.tracing)
        console.log ("Attempting to fetch '" + attemptUrl + "'");
  } else {
    if (ctx.tracing)
      console.log ("Attempting to fetch '" + attemptUrl + "' for", asset.name);
  }
  var fetch_promise = fetch_file_cb (attemptUrl);
  fetch_promise.then (handleFetchResponse);
} catch (exc) {
  console.error ("MONO_WASM: Error fetching " + attemptUrl, exc);
  attemptNextSource ();
}
```